delphin/util.py

```diff
@@ -272,3 +272,43 @@ def peek(self, n=0, skip=None, drop=False):
             self._buffer_fill(n + 1)
             datum = buffer[n]
         return datum
+
+
+# modified from https://www.python.org/dev/peps/pep-0263/#defining-the-encoding
+_encoding_symbol_re = re.compile(
+    r'^[ \t\f]*.*?coding[:=][ \t]*([-_.a-zA-Z0-9]+)', re.IGNORECASE)
+
+import codecs, io
+
+def detect_encoding(filename, default_encoding='utf-8', comment_char=b';'):
+    encoding = None
+    with io.open(filename, 'rb') as fh:
+        line1 = next(fh)
+        re_match1 = _encoding_symbol_re.search(line1)
+        line1_comment = line1.contains(comment_char)
+        if line1_comment and re_match1:
+            encoding = _check_encoding(re_match1.group(1))
+            if not encoding and line1 != "":
+                line2 = next(fh)
```

Comment:
I think I missed this when we met last week; I don't think we should examine the second line because none of our formats use [shebang](https://en.wikipedia.org/wiki/Shebang_(Unix)) interpreter directives. We should only check the first line for encoding directives.